src/docker/composebuilder/service_endpoints_test.go

```diff
@@ -0,0 +1,54 @@
+package composebuilder_test
+
+import (
+	"github.com/Originate/exosphere/src/config"
+	"github.com/Originate/exosphere/src/docker/composebuilder"
+	"github.com/Originate/exosphere/src/types"
+	"github.com/Originate/exosphere/test/helpers"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ServiceEndpoints", func() {
+	var appConfig types.AppConfig
+	var serviceRole string
+	var serviceConfigs map[string]types.ServiceConfig
+
+	var _ = BeforeEach(func() {
+		var err error
+		appDir := helpers.GetTestApplicationDir("simple")
+		appConfig, err = types.NewAppConfig(appDir)
+		Expect(err).NotTo(HaveOccurred())
+		serviceConfigs, err = config.GetServiceConfigs(appDir, appConfig)
+		Expect(err).NotTo(HaveOccurred())
+		serviceRole = "web"
+	})
+
+	It("compiles the proper local development endpoints", func() {
+		portReservation := types.NewPortReservation()
+		buildMode := composebuilder.BuildMode{
+			Type:        composebuilder.BuildModeTypeLocal,
+			Mount:       true,
+			Environment: composebuilder.BuildModeEnvironmentDevelopment,
+		}
+		s := composebuilder.NewServiceEndpoint(appConfig.Name, serviceRole, serviceConfigs[serviceRole], portReservation, buildMode)
+		endpoints := s.GetEndpointMappings()
+		Expect(endpoints["WEB_EXTERNAL_ORIGIN"]).To(Equal("http://localhost:3000"))
+		mapping := s.GetPortMappings()
+		Expect(mapping[0]).To(Equal("3000:8080"))
+	})
+
+	It("compiles the proper production development endpoints", func() {
+		portReservation := types.NewPortReservation()
+		buildMode := composebuilder.BuildMode{
+			Type:        composebuilder.BuildModeTypeLocal,
+			Mount:       true,
+			Environment: composebuilder.BuildModeEnvironmentProduction,
+		}
+		s := composebuilder.NewServiceEndpoint(appConfig.Name, serviceRole, serviceConfigs[serviceRole], portReservation, buildMode)
+		endpoints := s.GetEndpointMappings()
+		Expect(endpoints["WEB_EXTERNAL_ORIGIN"]).To(Equal("http://localhost:3000"))
+		mapping := s.GetPortMappings()
+		Expect(mapping[0]).To(Equal("3000:80"))
+	})
+})
```

Comment:
How about a test for deployments as well?

Reply:
Yep, that's covered here: https://github.com/Originate/exosphere/pull/805/files#diff-21817062a70daaf4c137792ca249e116R76